chkbit 4.2.1 (new formula)

laktak opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

chenrui333 commented on Formula/c/chkbit.rb:
```diff
@@ -0,0 +1,39 @@
+    assert_match(/usage: chkbit \[-h\].*/, shell_output("#{bin}/chkbit"))
+
+    # test operation
+    mkdir testpath/"foo" do
+      (testpath/"foo/one.txt").write <<~EOS
+        testing
+        testing
+        testing
+      EOS
+      system bin/"chkbit", "-u", testpath/"foo/"
+      assert_predicate testpath/"foo/.chkbit", :exist?
+    end
```
you dont need `foo` folder at all?

laktak replied:
I've removed the folder.

chenrui333 commented on Formula/c/chkbit.rb:
```diff
@@ -0,0 +1,39 @@
+  end
+
+  test do
+    assert_match(/usage: chkbit \[-h\].*/, shell_output("#{bin}/chkbit"))
```
can we check version in here?

laktak replied:
Yes, I've added a version check.

laktak commented:
@chenrui333  I think I've resolved all requested changes. Please let me know if you run into anything else.

chenrui333 commented on Formula/c/chkbit.rb:
```diff
@@ -0,0 +1,39 @@
+    assert_equal version.to_s, shell_output("#{bin}/chkbit --version").chomp
+
+    # test operation
+    mkdir testpath do
```
testpath is always there, you dont need to mk one

chenrui333 commented on Formula/c/chkbit.rb:
```diff
@@ -0,0 +1,39 @@
+
+  desc "Check your files for data corruption"
+  homepage "https://github.com/laktak/chkbit-py"
+  url "https://github.com/laktak/chkbit-py/archive/refs/tags/v4.2.1.tar.gz"
```
looks like this is not on pypi? (can you switch the source to pypi one, we'd prefer pypi over github tarball)

<img width="1198" alt="image" src="https://github.com/Homebrew/homebrew-core/assets/1580956/bdbcc88a-a862-47a6-9232-4149eb8c27e1">

laktak replied:
No problem, I've release it on pypi and switched the source.

chenrui333 commented on Formula/c/chkbit.rb:
```diff
@@ -0,0 +1,39 @@
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/chkbit --version").chomp
+
+    # test operation
+    mkdir testpath do
+      (testpath/"one.txt").write <<~EOS
+        testing
+        testing
+        testing
+      EOS
+      system bin/"chkbit", "-u", testpath
+      assert_predicate testpath/".chkbit", :exist?
```
```suggestion
    (testpath/"one.txt").write <<~EOS
      testing
      testing
      testing
    EOS

    system bin/"chkbit", "-u", testpath
    assert_predicate testpath/".chkbit", :exist?
```

chenrui333 commented:
can you also add `chkbit` into autobump.txt as well?

laktak commented:
@chenrui333 
> can you also add `chkbit` into autobump.txt as well?

ah, that is how this works :)

I've accepted your change, switched to pipy and added it to autobump.txt.

chenrui333 commented:
@laktak, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
